webpack/components/extensions/HostDetails/Tabs/ErrataTab.js

```diff
@@ -41,6 +43,11 @@ export const ErrataTab = () => {
   const response = useSelector(state => selectAPIResponse(state, HOST_ERRATA_KEY));
   const { results, ...metadata } = response;
   const status = useSelector(state => selectHostErrataStatus(state));
+
+  const selectAll = new SelectAllManager(results, metadata, useSet([]))
```

Comment:
I get what is happening here with the external class, but this is bringing external language principles into react, and obfuscating things more than is needed. 

I understand this is a principle of OOP that is appreciated in other languages, but in my experience, it usually ends in complicated difficult code to maintain when applied to React. 

If there is a simpler path that can have the code be contained within a component. I would prefer that.

Reply:
Working on it. Thanks!